HEC.FDA.Model/paireddata/PairedData.cs

```diff
@@ -232,62 +232,69 @@ public double integrate()
         /// </summary>
         public IPairedData multiply(IPairedData systemResponseFunction)
         {
-            double belowFragilityCurveValue = 0.0;
-            double aboveFragilityCurveValue = 1.0;
-            List<double> newXvals = new List<double>();
-            List<double> newYvals = new List<double>();
-            double buffer = .001; //buffer to define point just above and just below the multiplying curve.
-            if (Xvals[0] < systemResponseFunction.Xvals[0])
-            {
-                //cacluate no damage until the bottom of the fragility curve
-                double bottomStageOfSystemResponse = systemResponseFunction.Xvals[0];
-                foreach (double damageValue in Xvals)
-                {
-                    if (damageValue < bottomStageOfSystemResponse)
-                    {
-                        //set to zero
-                        newXvals.Add(damageValue);
-                        newYvals.Add(belowFragilityCurveValue);
-                    }
-                    else
-                    {
-                        //create a point on the curve just below the bottom of the levee at damage zero.
-                        newXvals.Add(bottomStageOfSystemResponse - buffer);
-                        newYvals.Add(belowFragilityCurveValue);
-                        break;
-                    }
-                }
-            }
-            // calculate damages for the range of the fragility curve
-            for (int index = 0; index < systemResponseFunction.Xvals.Count(); index++)
+            IPairedData newSystemResponse = EnsureBottomAndTopHaveCorrectProbabilities(systemResponseFunction);
+
+            List<double> newXvals = new List<double>(); //xvals are stages in the stage-damage function
+            List<double> newYvals = new List<double>(); //yvals are damage*prob(failure)
```

Comment:
This is probably fine, but arrays are going to be more efficient than lists. If we find this to be slow when we do profiling, we may have to come back and refactor. List seem elegant to me for this solution so I'd leave em for now.